background_scripts/background.js

```diff
@@ -257,19 +259,29 @@ chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
             type: sniff.type,
           };
         }
-        sniffs[msgTabId][sniff.domain]["details"].push({
-          inputField: {
-            fieldName: sniffDetails.fieldName,
-            value: sniffDetails.elValue,
-            xpath: sniffDetails.xpath,
-          },
-          timeStamp: sniffDetails.timeStamp,
-        });
+        const previouslySniffedDetails=sniffs[msgTabId][sniff.domain]["details"].find(d => d.inputField.fieldName===sniffDetails.fieldName && d.inputField.xpath===sniffDetails.xpath);
```

Comment:
Is it possible to do this look up without `find`? For instance by using `fieldName` and `xpath` as keys?
`sniffs[msgTabId][sniff.domain]["details"][sniffDetails.fieldName][sniffDetails.xpath]`, or something similar.

Reply:
Can we leave a space before and after the `=`:
 `const previouslySniffedDetails=sniffs...`  -> `const previouslySniffedDetails = sniffs...`

Reply:
tbh, I'm not aware of a better alternative to _find_ in this case, and I'm not familiar with any way to use keys that way.  
So please feel free to alter it however you feel works best.

Reply:
Sure. Thanks for the contribution. We'll take it from there.